packages/utilities/shared-utils/src/functions.ts

```diff
@@ -79,3 +81,125 @@ export function objectToDeps(obj: Extractable) {
     return "";
   }
 }
+
+export function debounce<F extends (...args: any[]) => void>(
+  func: F,
+  waitMilliseconds: number = 0,
+) {
+  let timeout: ReturnType<typeof setTimeout> | undefined;
+
+  return function (this: ThisParameterType<F>, ...args: Parameters<F>) {
+    const later = () => {
+      timeout = undefined;
+      func.apply(this, args);
+    };
+
+    if (timeout !== undefined) {
+      clearTimeout(timeout);
+    }
+
+    timeout = setTimeout(later, waitMilliseconds);
+  };
+}
+
+export function uniqBy<T>(arr: T[], iteratee: any) {
+  if (typeof iteratee === "string") {
+    iteratee = (item: T) => item[iteratee as keyof T];
+  }
+
+  return arr.filter((x, i, self) => i === self.findIndex((y) => iteratee(x) === iteratee(y)));
+}
+
+export const omit = <Obj, Keys extends keyof Obj>(obj: Obj, keys: Keys[]): Omit<Obj, Keys> => {
+  const res = Object.assign({}, obj);
+
+  keys.forEach((key) => {
+    delete res[key];
+  });
+
+  return res;
+};
+
+export const kebabCase = (s: string) => {
+  return s.replace(/([a-z])([A-Z])/g, "$1-$2").toLowerCase();
+};
+
+export const mapKeys = (
+  obj: Record<string, any>,
+  iteratee: (value: any, key: string) => any,
+): Record<string, any> => {
+  return Object.fromEntries(
+    Object.entries(obj).map(([key, value]) => [iteratee(value, key), value]),
+  );
+};
+
+export const get = (
+  object: Record<string, any>,
+  path: string | (string | number)[],
+  defaultValue?: any,
+): any => {
+  const keys = Array.isArray(path) ? path : path.replace(/\[(\d+)\]/g, ".$1").split(".");
+
+  let res: any = object;
+
+  for (const key of keys) {
+    res = res?.[key];
+
+    if (res === undefined) {
+      return defaultValue;
+    }
+  }
+
+  return res;
+};
+
+export const intersectionBy = <T>(...args: [...arrays: T[][], iteratee: Iteratee<T>]): T[] => {
```

Comment:
It would be helpful to have simple comments for the props and the result. For example, callers might not expect the function to throw an exception when args length is less than 2, so it’s important to document that.